app/src/components/OrderInformation.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { Component } from 'react';
+import { createStyles, WithStyles } from '@material-ui/core/styles';
+import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
+import Divider from '@material-ui/core/Divider';
+import { withStyles } from '@material-ui/styles';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import bigInt from 'big-integer';
+import { Message } from '../models/OrderBook';
+import { Styles } from '../styles/OrderInformation';
+import { MESSAGE_TYPE_ENUM } from '../constants/Enums';
+import {
+    convertNanosecondsUTCToCurrentTimezone, getLocalTimeString,
+    nanosecondsToString,
+    splitNanosecondEpochTimestamp,
+} from '../utils/date-utils';
+
+const styles = createStyles(Styles);
+
+interface Props extends WithStyles<typeof styles>{
+    orderId : number,
+    quantity: number,
+    lastModified: string,
+    createdOn: string,
+    price: number,
+    messages: Array<Message>,
+}
+
+interface State {
+    right: boolean,
+}
+
+class OrderInformation extends Component<Props, State> {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            right: true,
+        };
+    }
+
+    renderToggleDrawer(side, open) {
+        return event => {
+            if (event && event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
+                return;
+            }
+
+            this.setState({ right: true, [side]: open });
+        };
+    }
+
+    renderSlideList(side) {
```

Comment:
this custom render function already breaks down a part of the table, which is good. Can it be broken down even more, as it is still big.

rendering of the tablebody maybe in another function ?

Reply:
ya i think putting the message list table in the another function would be feasible!